service/src/main/java/uk/gov/hmcts/reform/fpl/validators/HasEPOGroundsValidator.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasEPOGrounds;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class HasEPOGroundsValidator implements ConstraintValidator<HasEPOGrounds, CaseData> {
+    @Override
+    public void initialize(HasEPOGrounds constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(CaseData caseData, ConstraintValidatorContext constraintValidatorContext) {
+        constraintValidatorContext.buildConstraintViolationWithTemplate(constraintValidatorContext
```

Comment:
Wouldn't it be more appropriate to add violation only if validator is about to return false?

I am even surprised it works correctly. Framework must be smart enough to disregard it when value is valid.

Reply:
I don't think it makes a difference since our return statement can return either true / false

Reply:
For it does not make sense to add constant to framework when there is no error. Have you checked documentation. They have the following example there:

```
        if ( object == null ) {
            return true;
        }

        boolean isValid;
        if ( caseMode == CaseMode.UPPER ) {
            isValid = object.equals( object.toUpperCase() );
        }
        else {
            isValid = object.equals( object.toLowerCase() );
        }

        if ( !isValid ) {
            constraintContext.disableDefaultConstraintViolation();
            constraintContext.buildConstraintViolationWithTemplate(
                    "{org.hibernate.validator.referenceguide.chapter06." +
                    "constraintvalidatorcontext.CheckCase.message}"
            )
            .addConstraintViolation();
        }

        return isValid;
```

Reply:
Ok, in our case we are returning a conditional boolean value. Would you be happy with just moving the constrainContext edits inside the loop towards the final return? Or should I use further if / else statements to have a clear failure case?